### ISSUE TITLE
Double tap to full screen issue

### ISSUE THREAD

**Issue by bvahedy** [AndBible/and-bible#1291]
When double tapping to go to full screen the single click dialog modal pops up.
I don't know if there is a way to avoid this issue, any ideas?

**Comment by agrogers** [comment]
Yes, that is a bit of pain. More unintended consequences :)

Adding a delay after the first click might be an option. But it will make
the click a little less responsive. There might be a sweet spot where it
can detect a double click but not feel slow?

I would probably just hide the bookmark action dialog when going full
screen after a double-click. If they want to keep it open then they can use
the '...' menu option.

Other thoughts here but i dont think they are too helpful either.
https://stackoverflow.com/questions/43444307/android-avoid-calling-single-tap-twice-for-double-tap-without-using-onsingletap


___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
I made now solution that will delay modal 300 ms to detect doubleclicks. Let's see if it works or if it's too much delay...

**Comment by timbze** [comment]
I am not a fan of using delay. Reason is I think user would use single click 100 times for 1 double click. To delay those 99 times is not a good sacrifice IMO 

I think better to just hide modal again if it's double click

If others don't agree then just close this again

**Comment by tuomas2** [comment]
I could remove that delay if double-tap to fullscreen option is disabled

**Comment by tuomas2** [comment]
Another click often hits the modal itself so it's tricky to do closing.

**Comment by tuomas2** [comment]
I feel whole double tap to fullscreen is not very convenient these days.

**Comment by agrogers** [comment]
I reckon give it a go with your delay. Let's see how it feels first. It
could be a simple solution and not too noticeable.

What about multi finger gestures to make it full screen? Even fewer people
would find it then :)

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
300ms is noticeable. And smaller delays did not work.

**Comment by tuomas2** [comment]
very disturbing...

**Comment by timbze** [comment]
Could perhaps disable double-tap and just use the 3-dot menu item for it. 

But then how to get out of full screen? I know some apps come out of full screen when swiping down from the top, but I think there should be some more obvious way to do it as well.

**Comment by bvahedy** [comment]
I like Andrew's idea about two finger swip expand gesture to entet full
screen.

**Comment by timbze** [comment]
The two finger expand / contract could be a good gesture for entering / exiting full screen.

Also, perhaps in full screen when tapping a verse, the modal should show an extra button for exit full screen

**Comment by tuomas2** [comment]
But can't change until 4.1, because of help texts.

**Comment by tuomas2** [comment]
What we could consider is to disable double-tap to fullscreen by default (users can re-toggle it and have modal delay then)

**Comment by timbze** [comment]
> What we could consider is to disable double-tap to fullscreen by default (users can re-toggle it and have modal delay then)

I would be ok with that. But we'd still need a way to exit full screen

**Comment by tuomas2** [comment]
Perhaps we should show toolbar when sliding down system toolbar somehow... Or auto-exit fullscreen if user does that. I'm not sure how to catch that event though.

**Comment by tuomas2** [comment]
code in hideSystemUI and showSystemUI seems mostly deprecated BTW

**Comment by bvahedy** [comment]
We can forget about help text for now, how about displaying a return icon?

**Comment by bvahedy** [comment]
I just discovered (ok I'm a bit slow) that if you double tap on a white
space anywhere on the screen you will enter the full screen mode without
bringing up the dialog. This works also in going back.
We could leave it the way it is assuming users arrange there text to have
empty white spaces on their screens. Any thoughts?

**Comment by howdev** [comment]
this is terrible.
double tap work as it should before version 4. It is not practical to find an empty area to double tap to fullscreen, this is an app with full screen of text. The bookmark dialog should be as previous versions, tap and hold to bring it up, then you have no problem with double tap fullscreen.
The tap once open bookmark dialog is problem in itself. Now you can't select more than one verse. Tap on the next verse, the dialog disappears. Instead of floating dialog blocking text, it should be fixed a the lower part of the screen, opens when tap and hold a verse

**Comment by bvahedy** [comment]
You can tap on the + icon on the dialog header to select multiple verses.
Also, you can disable double tap to full screen in the application preferences.

**Comment by howdev** [comment]
@bvahedy yes I know you can, but that is a tiny + and definitely not even good as tap+hold then tap on more verses

**Comment by agrogers** [comment]
You can still tap+hold+drag. That functionality hasn't changed. Once you get your selection choose whether you want to bookmark a whole verse or the exact selection. This video demonstrates highlighting part of a verse only - but you do the same thing for multiple verses: https://www.youtube.com/watch?v=PXdvFRLGcAA

I understand these sorts of changes can be frustrating... but they are not made without lots of discussion and forethought. In this case it was felt that what was lost was outweighed by what was gained. There is no question in my mind having used the new version for over a year that I would not go back. And I think new users to the app would probably agree.
